flang/lib/Lower/ConvertType.cpp

```diff
@@ -221,12 +223,22 @@ struct TypeBuilder {
         std::vector<std::pair<std::string, mlir::Type>> cs;
         auto &symbol = tySpec->typeSymbol();
         auto rec = fir::RecordType::get(context, toStringRef(symbol.name()));
```

Comment:
Question here: should derived type name be mangled, especially to include the kind parameters values ? Not asking to do this know, I am just wondering about the TODOs remaining in this code.

Reply:
They need to be. The internal name mangling scheme has a solution for that.

Reply:
I added a comment here.